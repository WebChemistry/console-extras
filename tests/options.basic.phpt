<?php declare(strict_types = 1);

use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Tester\Assert;
use WebChemistry\ConsoleExtras\Attribute\Description;
use WebChemistry\ConsoleExtras\Attribute\Option;
use WebChemistry\ConsoleExtras\ExtraCommand;

require __DIR__ . '/bootstrap.phpt';

$command = new class extends ExtraCommand {

	#[Option]
	#[Description('Basic option')]
	protected bool $basic;

	#[Option]
	protected ?string $require;

	#[Option]
	protected string $optional = 'default';

	#[Option(negatable: true)]
	protected bool $negatable = true;

	#[Option(shortcut: 's')]
	protected ?string $shortcut;

	protected function exec(InputInterface $input, OutputInterface $output): bool
	{
		$output->write(implode(',', array_map(fn ($value) => var_export($value, true), [
			$this->basic,
			$this->require,
			$this->optional,
			$this->negatable,
			$this->shortcut,
		])));

		return true;
	}

};

Assert::contains("false,NULL,'default',true,NULL", runConsoleApp($command));
Assert::contains("true,NULL,'foo',false,'bar'", runConsoleApp($command, [
	'--basic' => null,
	'--optional' => 'foo',
	'--no-negatable' => null,
	'-s' => 'bar',
]));
