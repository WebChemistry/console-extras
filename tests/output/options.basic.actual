
                                           
  The "optional" argument does not exist.  
                                           

cmd [--basic] [--require REQUIRE] [--optional [OPTIONAL]] [--negatable|--no-negatable] [-s|--shortcut SHORTCUT]

