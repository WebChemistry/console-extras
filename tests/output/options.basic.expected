false,NULL,'foo',true,NULL